src/openrct2-ui/drawing/engines/HardwareDisplayDrawingEngine.cpp

```diff
@@ -98,7 +101,40 @@ class HardwareDisplayDrawingEngine final : public X8DrawingEngine
             }
         }
 
-        _screenTexture = SDL_CreateTexture(_sdlRenderer, pixelFormat, SDL_TEXTUREACCESS_STREAMING, width, height);
+        sint32 scaleQuality = gConfigGeneral.scale_quality;
+        if (gConfigGeneral.use_nn_at_integer_scales &&
+            gConfigGeneral.window_scale == std::floor(gConfigGeneral.window_scale))
+        {
+            scaleQuality = 0;
+        }
+        if (scaleQuality == 3)
+        {
+            scaleQuality = 1;
+            smoothNN = true;
+        }
+        else
+        {
+            smoothNN = false;
+        }
+
+        if (smoothNN)
+        {
+            SDL_DestroyTexture(_scaledScreenTexture);
+
+            char scaleQualityBuffer[4];
+            snprintf(scaleQualityBuffer, sizeof(scaleQualityBuffer), "%u", scaleQuality);
+            SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "0");
+            _screenTexture = SDL_CreateTexture(_sdlRenderer, pixelFormat, SDL_TEXTUREACCESS_STREAMING, width, height);
+            SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, scaleQualityBuffer);
+
+            uint32 scale = std::ceil(gConfigGeneral.window_scale);
+            _scaledScreenTexture = SDL_CreateTexture(_sdlRenderer, pixelFormat, SDL_TEXTUREACCESS_TARGET,
```

Comment:
Shouldn't you release it somewhere?

Reply:
What do you mean with release exactly? I'm already destroying the texture a few lines above this (122) and I don't see anything I'm missing here.

Reply:
When resizing you handle it correctly, but when closing down, you don't release the texture. You should release it in the destructor as well.

Reply:
Like this?

Reply:
yes